integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -1112,4 +1108,25 @@ private String getNodePort() throws Exception {
               + "for the admin server in domain.");
     }
   }
+
+  private String getHostNameForCurl() throws Exception {
+    if (System.getenv("K8S_NODEPORT_HOST") != null) {
+      return System.getenv("K8S_NODEPORT_HOST");
+    } else {
+      // ExecResult result = ExecCommand.exec("hostname | awk -F. '{print $1}'");
+      ExecResult result1 =
+          ExecCommand.exec("kubectl get nodes -o=jsonpath='{range .items[0]}{.metadata.name}'");
+      if (result1.exitValue() != 0) {
+        throw new RuntimeException("FAILURE: Could not get K8s Node name");
+      }
+      ExecResult result2 =
+          ExecCommand.exec(
+              "nslookup " + result1.stdout() + " | grep \"^Name\" | awk '{ print $2 }'");
+      if (result2.stdout().trim().equals("")) {
```

Comment:
Maybe check exitValue of result2, just like you did for result1?   

Otherwise, LGTM.